test/quantization/fx/test_quantize_fx.py

```diff
@@ -5507,6 +5507,81 @@ def forward(self, x):
             out_ref = converted(inp)
 
             torch.testing.assert_allclose(out, out_ref)
+
+    def test_disable_output_fake_quant(self):
+
+        class Linear(torch.nn.Module):
+            def __init__(self):
+                super().__init__()
+                self.w = torch.ones(5, 5)
+                self.b = torch.zeros(5)
+
+
+            def forward(self, x):
+                return torch.nn.functional.linear(x, self.w, self.b)
+
+        class M(torch.nn.Module):
+            def __init__(self):
+                super().__init__()
+                self.mods1 = torch.nn.Sequential(
+                    Linear(),
+                    Linear()
+                )
+                self.mods2 = Linear()
+                self.mods3 = Linear()
+
+            def forward(self, x):
+                x = self.mods1(x)
+                x = torch.add(x, 4)
+                x = self.mods2(x)
+                y = torch.add(x, 2)
+                z = torch.mul(x, 5)
+                a = self.mods3(y)
+                return a, z
+        devices = ["cpu", "cuda"] if torch.cuda.is_available() else ["cpu"]
+
+        for device in devices:
+            model = M().train()
+            model.to(device)
+            qconfig = torch.quantization.get_default_qat_qconfig("fbgemm")
+            qconfig_dict = {
+                "": None,
+                "object_type": [
+                    (torch.nn.functional.linear, qconfig),
+                    (torch.nn.functional.relu, qconfig),
+                ],
+            }
+            prepared = torch.quantization.quantize_fx.prepare_qat_fx(model, qconfig_dict)
+            prepared.to(device)
+            modules = dict(prepared.named_modules())
+
+            def check_all_users_not_linear(node):
+                for user, _ in node.users.items():
+                    if user.op == 'call_function' and user.target == torch.nn.functional.linear:
+                        return False
+                return True
+
+
+            for node in prepared.graph.nodes:
+                if node.op == 'call_module' and torch.quantization.is_activation_post_process(modules[str(node.target)]):
+                    # get the users
+                    if check_all_users_not_linear(node):
```

Comment:
is this what we are planning to do in practice?

Reply:
this is one example (specific to linear) of how the output fake-quant can be disabled for a certain FakeQuantize module.